src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs

```diff
@@ -284,19 +284,10 @@ private void AnalyzeMethodOverloads(OperationAnalysisContext context, IMethodSym
                             return;
                         }
 
-                        if (!_xmlReaderSettingsEnvironments.TryGetValue(settingsSymbol, out XmlReaderSettingsEnvironment env))
-                        {
-                            // symbol for settings is not found => passed in without any change => assume insecure
-                            Diagnostic diag = Diagnostic.Create(
-                                RuleDoNotUseInsecureDtdProcessing,
-                                expressionSyntax.GetLocation(),
-                                SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                    nameof(MicrosoftNetFrameworkAnalyzersResources.XmlReaderCreateInsecureInputMessage)
-                                )
-                            );
-                            context.ReportDiagnostic(diag);
-                        }
-                        else if (!env.IsDtdProcessingDisabled && !(env.IsSecureResolver && env.IsMaxCharactersFromEntitiesLimited))
+                        // If we have no XmlReaderSettingsEnvironment, then we don't know.
```

Comment:
Just curious: what does legacy FxCop do for this case?

Reply:
Haven't looked too deeply, but it doesn't generate a warning on the unit test I added.